serve/filters/filter/tx.go

```diff
@@ -0,0 +1,203 @@
+package filter
+
+import (
+	"sort"
+
+	"github.com/gnolang/gno/tm2/pkg/bft/types"
+)
+
+// filterPriority defines the priority of a filter condition.
+// The higher the priority, the earlier the condition is applied.
+//
+// This concept is borrowed from operator precedence.
+type filterPriority int
+
+const (
+	HeightPriority filterPriority = iota // highest priority
+	IndexPriority
+	GasUsedPriority
+	GasWantedPriority // lowest priority
+)
+
+type condition struct {
+	filter   func(*types.TxResult) bool
+	priority filterPriority
+}
+
+// TxFilter holds a slice of transaction results.
+// It provides methods to manipulate and query the transactions.
+type TxFilter struct {
+	*baseFilter
+	// txrs represents the transactions in the filter.
+	txrs []*types.TxResult
+	// conditions holds the filtering conditions.
+	conditions []condition
+}
+
+// NewTxFilter creates a new TxFilter object.
+func NewTxFilter() *TxFilter {
+	return &TxFilter{
+		baseFilter: newBaseFilter(TxFilterType),
+		txrs:       make([]*types.TxResult, 0),
+		conditions: make([]condition, 0),
+	}
+}
+
+// GetHashes iterates over all transactions in the filter and returns their hashes.
+//
+// It appends `nil` to the result slice if the transaction or its content is `nil`.
+// This ensures that the length og the returned slice matches the number of transactions in the filter.
+func (tf *TxFilter) GetHashes() [][]byte {
+	tf.Lock()
+	defer tf.Unlock()
+
+	hashes := make([][]byte, 0, len(tf.txrs))
+
+	for _, txr := range tf.txrs {
+		if txr == nil || txr.Tx == nil {
```

Comment:
This situation is just never possible, we don't have to cover cases when the transaction object is being added is `nil` or malformed

Reply:
I added it just in case, but I'll remove it.